clients/src/main/java/org/apache/kafka/common/metrics/Gauge.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
+/**
+ * A non-measurable gauge value that can be registered as a metric.
+ * This is separate from {@link Measurable} to avoid breaking the public API.
+ */
+public interface Gauge<T> {
+
+    /**
+     * Returns the current value associated with this gauge
+     */
+    public T value();
```

Comment:
Is there a reason why this doesn't take the same parameters as `Measurable`? We could potentially introduce a super-interface of `Measurable` that returns `T` with a different method name and have `Measurable` implement that by calling `measure`. But we should wait until Java 8. Worth maybe adding a comment stating this.

Reply:
I used the signature for yammer Gauge since gauge value doesn't depend on current time or config. But I have changed it for the reason you stated above and added a comment.